pinot-core/src/main/java/org/apache/pinot/core/util/SegmentProcessorAvroUtils.java

```diff
@@ -74,7 +74,10 @@ public static GenericData.Record convertGenericRowToAvroRecord(GenericRow generi
   public static Schema convertPinotSchemaToAvroSchema(org.apache.pinot.spi.data.Schema pinotSchema) {
     SchemaBuilder.FieldAssembler<org.apache.avro.Schema> fieldAssembler = SchemaBuilder.record("record").fields();
 
-    for (FieldSpec fieldSpec : pinotSchema.getAllFieldSpecs()) {
+    List<FieldSpec> orderedFieldSpecs = pinotSchema.getAllFieldSpecs().stream()
+        .sorted()
```

Comment:
Good catch! Here we want to order the fields by name. `FieldSpec` itself is not comparable

Reply:
Thanks for the comment @Jackie-Jiang, I overlooked that. I've fixed the sorting now.

Reply:
Can you please also fix the compile issue? I believe some classes are not imported

Reply:
My bad @Jackie-Jiang. I've had some problems getting the project to work with maven lately, I'll get this sorted.